go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/openshift/hypershift
 
-go 1.23.0
+go 1.23.4
```

Comment:
One way to drop that dependency is to release 0.12.1 in CAPI which relies has 1.23.0 instead of 1.23.4.
I think it's better practice to not hard-code the patch version of go in go.mod AFIK.

Reply:
okay, I just fixed it.